pkgs/applications/networking/mailreaders/claws-mail/default.nix

```diff
@@ -1,104 +1,187 @@
-      "--disable-pgpmime-plugin"
+    [
+      "--disable-manual"   # Missing docbook-tools, e.g., docbook2html
+      "--disable-compface" # Missing compface library
+      "--disable-jpilot"   # Missing jpilot library
+
+      "--disable-gdata-plugin" # Complains about missing libgdata, even when provided
+      "--disable-fancy-plugin" # Missing libwebkit-1.0 library
     ]
-    ++ optional (!enablePluginArchive) "--disable-archive-plugin"
-    ++ optional (!enablePluginLitehtmlViewer) "--disable-litehtml_viewer-plugin"
-    ++ optional (!enablePluginPdf) "--disable-pdf_viewer-plugin"
-    ++ optional (!enablePluginPython) "--disable-python-plugin"
-    ++ optional (!enablePluginRavatar) "--disable-libravatar-plugin"
-    ++ optional (!enablePluginRssyl) "--disable-rssyl-plugin"
-    ++ optional (!enablePluginSmime) "--disable-smime-plugin"
-    ++ optional (!enablePluginSpamassassin) "--disable-spamassassin-plugin"
-    ++ optional (!enablePluginSpamReport) "--disable-spam_report-plugin"
-    ++ optional (!enablePluginVcalendar) "--disable-vcalendar-plugin"
-    ++ optional (!enableSpellcheck) "--disable-enchant";
+    ++
+    (map (e: strings.enableFeature (lists.head e) (lists.last e)) [
+      [ enableLibSM "libsm" ]
+      [ enableGnuTLS "gnutls" ]
+      [ enableEnchant "enchant" ]
+      [ enableDbus "dbus" ]
+      [ enableLdap "ldap" ]
+      [ enableNetworkManager "networkmanager" ]
+      [ enableLibetpan "libetpan" ]
+      [ enableValgrind "valgrind" ]
+      [ enableSvg "svg" ]
 
-  enableParallelBuilding = true;
+      [ enablePluginAcpiNotifier "acpi_notifier-plugin" ]
+      [ enablePluginAddressKeeper "address_keeper-plugin" ]
+      [ enablePluginArchive "archive-plugin" ]
+      [ enablePluginAttRemover "att_remover-plugin" ]
+      [ enablePluginAttachWarner "attachwarner-plugin" ]
+      [ enablePluginBogofilter "bogofilter-plugin" ]
+      [ enablePluginBsfilter "bsfilter-plugin" ]
+      [ enablePluginClamd "clamd-plugin" ]
+      [ enablePluginDillo "dillo-plugin" ]
+      [ enablePluginFetchInfo "fetchinfo-plugin" ]
+      [ enablePluginLibravatar "libravatar-plugin" ]
+      [ enablePluginLitehtmlViewer "litehtml_viewer-plugin" ]
+      [ enablePluginMailmbox "mailmbox-plugin" ]
+      [ enablePluginManageSieve "managesieve-plugin" ]
+      [ enablePluginNewMail "newmail-plugin" ]
+      [ enablePluginNotification "notification-plugin" ]
+      [ enablePluginPdfViewer "pdf_viewer-plugin" ]
+      [ enablePluginPerl "perl-plugin" ]
+      [ enablePluginPython "python-plugin" ]
+      [ enablePluginPgp "pgpcore-plugin" ]
+      [ enablePluginPgp "pgpmime-plugin" ]
+      [ enablePluginPgp "pgpinline-plugin" ]
+      [ enablePluginRssyl "rssyl-plugin" ]
+      [ enablePluginSmime "smime-plugin" ]
+      [ enablePluginSpamassassin "spamassassin-plugin" ]
+      [ enablePluginSpamReport "spam_report-plugin" ]
+      [ enablePluginTnefParse "tnef_parse-plugin" ]
+      [ enablePluginVcalendar "vcalendar-plugin" ]
```

Comment:
This should get a lot cleaner and nicer with some Nix attributes and functions applied. Start with this up above:

```nix
pluginsDeps = {
  acpi_notifier = [];
  archive = [ /*..*/ ];
  att_remover = [];
  # ...
};
```

Then use [`mapAttributesToList`](https://nixos.org/manual/nixpkgs/stable/#function-library-lib.attrsets.mapAttrsToList) to map the plugins enabled, to list of `configureFlags` (with `enableFeature`) and a list of `pluginsDeps`.

Reply:
You've got a point there. I will take a look at this later / tomorrow. Thanks!

Reply:
@oxzi Are you still planning to do this?

Reply:
Perhaps it'd be worth re-reviewing this, for the sake of the update, and revisit in the future the idea at using such Nix attributes..